ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/atomicaction/AtomicActionAsyncTest.java

```diff
@@ -49,6 +55,42 @@ public void testAbort() throws Exception {
     }
 
     @Test
+    public void testAbortAsyncAC() throws Exception {
+        boolean[] testAbortAsyncAC = new boolean[]{true};
+        int[] called = new int[1];
+        int[] rollback = new int[1];
+
+        AtomicActionTestBase.init(false);
+        arjPropertyManager.getCoordinatorEnvironmentBean().setAsyncAfterSynchronization(true);
+
+        final Thread thread = Thread.currentThread();
+        executeTest(false, ActionStatus.ABORTED, new SyncRecord[]{new SyncRecord() {
+            @Override
+            public boolean afterCompletion(int status) {
+                if (status == ActionStatus.ABORTED) {
+                    rollback[0]++;
+                }
+                testAbortAsyncAC[0] = testAbortAsyncAC[0] && (Thread.currentThread() != thread);
```

Comment:
Good point, I could remove the thread checks to be fair. The main thing is that without the patch you get an exception